src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/DueDateIntervalReCalculator.java

```diff
@@ -2,146 +2,106 @@
-                                           .orElse(List.of(DEFAULT_NON_WORKING_CALENDAR)))
-            .dueDateNonWorkingDaysOfWeek(dueDateProperties.stream()
-                                             .filter(r -> r.getName().getValue().equals(
-                                                 DUE_DATE_NON_WORKING_DAYS_OF_WEEK))
-                                             .filter(r -> r.getCanReconfigure().getValue().booleanValue() == TRUE)
-                                             .reduce((a, b) -> b)
-                                             .map(ConfigurationDmnEvaluationResponse::getValue)
-                                             .map(CamundaValue::getValue)
-                                             .map(s -> s.split(","))
-                                             .map(a -> Arrays.stream(a).map(String::trim).toArray(String[]::new))
-                                             .map(Arrays::asList)
-                                             .orElse(List.of()))
-            .dueDateSkipNonWorkingDays(dueDateProperties.stream()
-                                           .filter(r -> r.getName().getValue()
-                                               .equals(DUE_DATE_SKIP_NON_WORKING_DAYS))
-                                           .filter(r -> r.getCanReconfigure().getValue().booleanValue() == TRUE)
-                                           .reduce((a, b) -> b)
-                                           .map(ConfigurationDmnEvaluationResponse::getValue)
-                                           .map(CamundaValue::getValue)
-                                           .map(Boolean::parseBoolean)
-                                           .orElse(false))
-            .dueDateMustBeWorkingDay(dueDateProperties.stream()
-                                         .filter(r -> r.getName().getValue().equals(DUE_DATE_MUST_BE_WORKING_DAYS))
-                                         .filter(r -> r.getCanReconfigure().getValue().booleanValue() == TRUE)
-                                         .reduce((a, b) -> b)
-                                         .map(ConfigurationDmnEvaluationResponse::getValue)
-                                         .map(CamundaValue::getValue)
-                                         .orElse(DUE_DATE_MUST_BE_WORKING_DAY_NEXT))
-            .dueDateTime(dueDateProperties.stream()
-                             .filter(r -> r.getName().getValue().equals(DUE_DATE_TIME))
-                             .filter(r -> r.getCanReconfigure().getValue().booleanValue() == TRUE)
-                             .reduce((a, b) -> b)
-                             .map(ConfigurationDmnEvaluationResponse::getValue)
-                             .map(CamundaValue::getValue)
-                             .orElse(DEFAULT_DUE_DATE_TIME))
+    private DateTypeIntervalData readDueDateOriginFields(List<ConfigurationDmnEvaluationResponse> dueDateProperties) {
```

Comment:
This method is same as the readDueDateOriginFields in the super class other than checking the canReconfigure flag. You can pass it as a parameter to the method. Please see how its done in this pr https://github.com/hmcts/wa-task-management-api/pull/915